File: Hydra/ActivityMapController.m
//
//  ActivityMapViewController.m
//  Hydra
//
//  Created by Pieter De Baets on 27/08/13.
//  Copyright (c) 2013 Zeus WPI. All rights reserved.
//

#import "ActivityMapController.h"
#import "Hydra-Swift.h"

@interface SimpleMapAnnotation : NSObject<MKAnnotation>

@property (nonatomic, readonly) CLLocationCoordinate2D coordinate;
@property (nonatomic, readonly, copy) NSString *title;

- (id)initWithCoordinate:(CLLocationCoordinate2D)coordinate title:(NSString *)title;

@end

@interface ActivityMapController ()

@property (nonatomic, strong) Activity *activity;

@end

@implementation ActivityMapController

- (id)initWithActivity:(Activity *)activity
{
    if (self = [super init]) {
        self.activity = activity;
    }
    return self;
}

- (void)loadMapItems
{
    CLLocationCoordinate2D coordinate = CLLocationCoordinate2DMake(self.activity.latitude,
                                                                   self.activity.longitude);
    SimpleMapAnnotation *mapItem = [[SimpleMapAnnotation alloc] initWithCoordinate:coordinate
                                                                             title:self.activity.location];
    [self.mapView addAnnotation:mapItem];
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.title = @"Kaart";
}

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    GAI_Track(@"Activiteit > ... > Kaart");
}

@end

@implementation SimpleMapAnnotation

- (id)initWithCoordinate:(CLLocationCoordinate2D)coordinate title:(NSString *)title {
    if (self = [super init]) {
        _coordinate = coordinate;
        _title = title;
    }
    return self;
}

@end

File: Hydra/ActivityMapController.swift
//
//  ActivityMapController.swift
//  Hydra
//
//  Created by Feliciaan De Palmenaer on 06/07/2017.
//  Copyright © 2017 Zeus WPI. All rights reserved.
//

import Foundation
import MapKit

class ActivityMapController: MapViewController {
    
    let activity: Activity
    
    init(activity: Activity) {
        self.activity = activity
        super.init()
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func loadMapItems() {
        let annotation = SimpleMapAnnotation(latitude: activity.latitude, longitude: activity.longitude, title: activity.title, subtitle: activity.location)
        self.mapView.addAnnotation(annotation)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Kaart"
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        //GAI_TRACK("Activiteit > ... > Kaart")
    }
}

class SimpleMapAnnotation: NSObject, MKAnnotation {
    let coordinate: CLLocationCoordinate2D
    let title: String?
    let subtitle: String?
    
    init(coordinate: CLLocationCoordinate2D, title: String?, subtitle: String?) {
        self.coordinate = coordinate
        self.title = title
        self.subtitle = subtitle
    }
    
    convenience init(latitude: CLLocationDegrees, longitude: CLLocationDegrees, title: String?, subtitle: String?) {
        self.init(coordinate: CLLocationCoordinate2D(latitude: latitude, longitude: longitude), title: title, subtitle: subtitle)
    }
}
